string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,179 @@
+  var colonChecker = '';
+  for (var i = 0; i < title.length; i++) {
+    if (title[i] !== ' ') {
+      tempString += title[i].toLowerCase();
+      if (title[i] === '-') {
+        dashDetect = true;
+      }
+    } else {
+      if (tempString.length > 3) {
+        var fourWordMore = strFourMore(tempString, dashDetect, colonChecker);
+        if (titleString.length === 0) {
+          titleString += fourWordMore[0].toUpperCase();
+          for (var j = 1; j < fourWordMore.length; j++) {
+            titleString += fourWordMore[j];
+          }
+        } else {
+          titleString += fourWordMore;
+        }
+        titleString += ' ';
+        colonChecker = titleString[titleString.length - 2];
+        tempString = '';
+        dashDetect = false;
+      } else if (tempString.length <= 3) {
+        var threeWordLess = strThreeLess(tempString, dashDetect, colonChecker);
+        if (titleString.length === 0) {
+          titleString += threeWordLess[0].toUpperCase();
+          for (var k = 1; k < threeWordLess.length; k++) {
+            titleString += threeWordLess[k];
+          }
+        } else {
+          titleString += threeWordLess;
+        }
+        titleString += ' ';
+        colonChecker = titleString[titleString.length - 2];
+        tempString = '';
+      }
+    }
+  }
+  if (tempString.length > 3) {
+    fourWordMore = strFourMore(tempString, dashDetect, colonChecker);
+    titleString += fourWordMore;
+  } else if (tempString.length <= 3) {
+    threeWordLess = strThreeLess(tempString, dashDetect, colonChecker);
+    titleString += threeWordLess;
+  }
+  // console.log('final:', titleString)
+  return titleString;
+}
+
+function strFourMore(string, dashDetect, colonChecker) {
+  var temp = '';
+  if (dashDetect === true) {
+    temp += string[0].toUpperCase();
+    for (var i = 1; i < string.length; i++) {
+      if (string[i - 1] === '-') {
+        temp += string[i].toUpperCase();
+      } else {
+        temp += string[i];
+      }
+    }
+    return temp;
+  }
+  if (string === 'javascript') {
+    temp += 'JavaScript';
+    return temp;
+  }
+  if (string === 'javascript:') {
+    temp += 'JavaScript:';
+    return temp;
+  }
+  temp += string[0].toUpperCase();
+  for (var j = 1; j < string.length; j++) {
+    temp += string[j];
+  }
+  return temp;
+}
+
+function strThreeLess(string, dashDetect, colonChecker) {
+  var conjunctionsWords = ['and', 'or', 'not', 'but', 'as', 'at', 'by', 'for', 'in', 'of', 'on', 'per', 'to', 'the'];
+  var temp = '';
+  if (string === 'api') {
+    temp += string.toUpperCase();
+    return temp;
+  }
+  if (colonChecker === ':') {
+    temp += string[0].toUpperCase();
+    for (var i = 1; i < string.length; i++) {
+      temp += string[i];
+    }
+    return temp;
+  }
+  if (conjunctionsWords.indexOf(string) !== -1) {
+    temp += string;
+    return temp;
+  }
+  temp += string[0].toUpperCase();
+  for (var j = 1; j < string.length; j++) {
+    temp += string[j];
+  }
```

Comment:
You should make this into one conditional block with a set of `else if` statements, then just `return temp` below the conditional. No matter which block runs, you always need to `return` the resultant value, so you should just generate the proper value, and `return` it below the conditional.